packages/doc/docs/api-building/configuration/data-api-schema.mdx

```diff
@@ -0,0 +1,158 @@
+    validators:
+      - uuid
+      - required
+profiles:
+  - pg # replace this with your profile name
+```
+
+VulcanSQL will read the Data API Schema and know what the SQL files mapping it, then auto-generated APIs that map to the user-written SQL files.
+
+Below is the introduction of each field in the structure.
+
+# API Schema Fields
+
+### `urlPath` Field
+
+This is the Data API endpoint URL path. you could add the path parameter hereby `:[param-name]`, format, but when you add the path parameter, the `feildName` should also have the path parameter, and `fieldIn` should be `path`, if you forgot, VulcanSQL will auto-generated it to make it work.
+
+```yaml
+urlPath: /users/:id
+request:
+  - fieldName: id
+    fieldIn: path
+    ....
+```
+
+After API generated, VulcanSQL will add a prefix `/api` to distinguish Data API endpoints and other Non-Data API endpoints. For the above sample, the request url actually should be `/api/users/:id`.
+
+Otherwise, if you not set `urlPath`, VulcanSQL will use your API Schema filename to be url, e.g: API schema filename is `user.yaml` , then the `urlPath` will be `/user` .
+
+### `templateSource` Field
+
+The`templateSource` field could let VulcanSQL know what the SQL file this API will like to map for querying it and response data. It should be SQL filename ( not need file extension ).
+
+```yaml
+- sqls
+  - user.yaml
+  - user.sql
+
+# user.yaml
+urlPath: /users/:id
+templateSource: user
+...
+```
+
+### `request` Fields
+
+The `request` define what the request parameters would like to send with the API request, if you would like to have parameters to send the request, you should define the parameter with at least `fieldName`, `fieldIn` two fields.
+
+- `fieldName` - The field name of the parameter of the API request, but the parameter field is optional, so you could send the request with the parameter optional. However, if your parameter is a path parameter e.g: `/users/:id`, then VulcanSQL will make the path parameter **required**.
+- `fieldIn` - Where the parameter field is put in with the API request. It could be `path`, `query`, and `header`. If the `fieldIn` is the `path`, then you should also define the path parameter in `urlPath` . below is the sample:
+
+  ```yaml
+  urlPath: users
+  request:
+    - fieldName: gender
+      fieldIn: query
+      ...
+    - fieldName: age
+      fieldIn: query
+  ```
+
+  The request could be below 4 cases:
+
+  ```bash
+  /api/users
+  /api/users?gender=male
+  /api/users?age=18
+  /api/users?gender=male&age=18
+
+  ```
+
+- `type` : The `type` could let VulcanSQL know the data is what type for handing it. The type could be `string`, `number`, and `boolean`.
+  If not set the field, the default is `string` type. Otherwise, sometimes if your SQL query should use the correct data type to do the filter condition, e.g: `where age > 18` , then if you don't set the `type` to be a `number` for `age`, then the query filter may get an incorrect result.
```

Comment:
Thanks so much for pointing out, maybe we could change the statment `Otherwise, sometimes if your SQL query should use the correct data type to do the filter condition, e.g: `where age > 18` , then if you don't set the `type` to be a `number` for `age`, then the query filter may get an incorrect result.` to your comment: `But don't worry. In most time, the driver is smart enough to handle parameters types, so it might take more effect on validators.`